packages/api/endpoints/distribution.js

```diff
@@ -85,94 +60,129 @@ function getSignedUrl(s3Client, Bucket, Key, username) {
   return parsedSignedUrl.toString();
 }
 
-/**
- * Given a an API Gateway request, return either the proxy path parameter or
- *   the state query string parameter
- *
- * @param {Object} request - an API Gatway request object
- * @returns {string|undefined} a granule location
- */
-function getGranuleLocationFromRequest(request) {
-  return get(request, 'pathParameters.proxy')
-    || get(request, 'queryStringParameters.state');
+function getAccessTokenFromRequest(request) {
+  const accessTokenCookie = getCookie(request, 'accessToken');
+
+  return accessTokenCookie ? accessTokenCookie.value : null;
 }
 
-/**
- * Return the username associated with an OAuth2 authorization code
- *
- * @param {EarthdataLoginClient} earthdataLoginClient - an Earthdata Login Client
- * @param {string} authorizationCode - the OAuth2 authorization code to use
- * @returns {string} an Earthdata username
- */
-async function getUsernameFromAuthorizationCode(earthdataLoginClient, authorizationCode) {
+function isRedirectRequest(request) {
+  return request.resource === '/redirect';
+}
+
+function isAccessTokenExpired(accessTokenRecord) {
+  return accessTokenRecord.expirationTime < Date.now();
+}
+
+async function handleRedirectRequest(params = {}) {
   const {
-    username
-  } = await earthdataLoginClient.getAccessToken(authorizationCode);
+    accessTokenModel,
+    authClient,
+    distributionUrl,
+    request
+  } = params;
+
+  const { code, state } = request.queryStringParameters;
+
+  const getAccessTokenResponse = await authClient.getAccessToken(code);
```

Comment:
So this handles the redirect from the OAuth provider after authentication there?

Reply:
Correct.